From bbcd5fc8782a1a26fdbf60563b9973093e171ebf Mon Sep 17 00:00:00 2001
From: Alexandre Rostovtsev <tetromino@gmail.com>
Date: Sat, 8 Oct 2011 03:31:56 -0400
Subject: [PATCH] Disable code not relevant for Gentoo

Disable code that is only relevant for an RPM-based distro or that
requires additional bugs.gentoo.org infrastructure support.
---
 configure.ac                      |    1 -
 doc/Makefile.am                   |    2 --
 src/daemon/Makefile.am            |   23 +----------------------
 src/daemon/abrt_event.conf        |    2 +-
 src/plugins/Makefile.am           |   16 ----------------
 src/plugins/abrt-action-list-dsos |   18 ++++--------------
 src/plugins/ccpp_event.conf       |    7 +++----
 src/plugins/koops_event.conf      |    4 ++--
 src/plugins/python_event.conf     |    6 +++---
 9 files changed, 14 insertions(+), 65 deletions(-)

diff --git a/configure.ac b/configure.ac
index 6b0732c..a340b55 100644
--- a/configure.ac
+++ b/configure.ac
@@ -60,7 +60,6 @@ PKG_CHECK_MODULES([GTK], [gtk+-2.0])
 PKG_CHECK_MODULES([GLIB], [glib-2.0 >= 2.21])
 PKG_CHECK_MODULES([DBUS], [dbus-1])
 PKG_CHECK_MODULES([LIBXML], [libxml-2.0])
-PKG_CHECK_MODULES([RPM], [rpm])
 PKG_CHECK_MODULES([LIBNOTIFY], [libnotify])
 PKG_CHECK_MODULES([NSS], [nss])
 PKG_CHECK_MODULES([BTPARSER], [btparser])
diff --git a/doc/Makefile.am b/doc/Makefile.am
index 551bc65..51fc97d 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -11,12 +11,10 @@ MAN1_TXT += abrt-action-list-dsos.txt
 MAN1_TXT += abrt-retrace-client.txt
 MAN1_TXT += abrt-handle-upload.txt
 MAN1_TXT += abrt-server.txt
-MAN1_TXT += abrt-action-save-package-data.txt
 MAN1_TXT += abrt-install-ccpp-hook.txt
 
 MAN5_TXT =
 MAN5_TXT += abrt.conf.txt
-MAN5_TXT += abrt-action-save-package-data.conf.txt
 
 MAN8_TXT =
 MAN8_TXT += abrtd.txt
diff --git a/src/daemon/Makefile.am b/src/daemon/Makefile.am
index 82ba93b..07ff6c0 100644
--- a/src/daemon/Makefile.am
+++ b/src/daemon/Makefile.am
@@ -6,9 +6,6 @@ dist_eventsconf_DATA = \
 bin_SCRIPTS = \
     abrt-handle-upload
 
-bin_PROGRAMS = \
-    abrt-action-save-package-data
-
 sbin_PROGRAMS = \
     abrtd \
     abrt-server
@@ -60,30 +57,12 @@ abrt_handle_event_LDADD = \
     ../lib/libabrt.la \
     $(LIBREPORT_LIBS)
 
-abrt_action_save_package_data_SOURCES = \
-    rpm.h rpm.c \
-    abrt-action-save-package-data.c
-abrt_action_save_package_data_CPPFLAGS = \
-    -I$(srcdir)/../include \
-    -I$(srcdir)/../lib \
-    -DCONF_DIR=\"$(CONF_DIR)\" \
-    $(GLIB_CFLAGS) \
-    $(LIBREPORT_CFLAGS) \
-    -D_GNU_SOURCE \
-    -Wall -Wwrite-strings -Werror
-abrt_action_save_package_data_LDADD = \
-    $(RPM_LIBS) \
-    $(LIBREPORT_LIBS) \
-    ../lib/libabrt.la
-
 dbusabrtconfdir = ${sysconfdir}/dbus-1/system.d/
 dist_dbusabrtconf_DATA = dbus-abrt.conf
 
 daemonconfdir = $(CONF_DIR)
 dist_daemonconf_DATA = \
-    abrt.conf \
-    abrt-action-save-package-data.conf \
-    gpg_keys
+    abrt.conf
 
 comredhatabrtservicedir = ${datadir}/dbus-1/system-services
 dist_comredhatabrtservice_DATA = com.redhat.abrt.service
diff --git a/src/daemon/abrt_event.conf b/src/daemon/abrt_event.conf
index dec0f0e..d2e48ff 100644
--- a/src/daemon/abrt_event.conf
+++ b/src/daemon/abrt_event.conf
@@ -44,7 +44,7 @@
 
 
 # Determine in which package/component the crash happened (if not yet done):
-EVENT=post-create component= remote!=1
+#EVENT=post-create component= remote!=1
         abrt-action-save-package-data
 
 
diff --git a/src/plugins/Makefile.am b/src/plugins/Makefile.am
index 9a26877..ac0033d 100644
--- a/src/plugins/Makefile.am
+++ b/src/plugins/Makefile.am
@@ -1,7 +1,6 @@
 -include ../../config.mak
 
 bin_SCRIPTS = \
-    abrt-action-install-debuginfo \
     abrt-action-analyze-core \
     abrt-action-analyze-vmcore \
     abrt-action-list-dsos
@@ -16,8 +15,6 @@ bin_PROGRAMS = \
     abrt-action-analyze-backtrace \
     abrt-retrace-client
 
-libexec_PROGRAMS = abrt-action-install-debuginfo-to-abrt-cache
-
 #dist_pluginsconf_DATA = Python.conf
 
 eventsdir = $(EVENTS_DIR)
@@ -49,7 +46,6 @@ dist_eventsconf_DATA = \
 
 
 PYTHON_FILES = \
-    abrt-action-install-debuginfo.in \
     abrt-action-list-dsos \
     abrt-action-analyze-core
 
@@ -157,18 +153,6 @@ abrt_action_analyze_backtrace_LDADD = \
     $(LIBREPORT_LIBS) \
     $(BTPARSER_LIBS)
 
-abrt_action_install_debuginfo_to_abrt_cache_SOURCES = \
-    abrt-action-install-debuginfo-to-abrt-cache.c
-abrt_action_install_debuginfo_to_abrt_cache_CPPFLAGS = \
-    -I$(srcdir)/../include \
-    -I$(srcdir)/../lib \
-    -D_GNU_SOURCE \
-    $(LIBREPORT_CFLAGS) \
-    -Wall -Wwrite-strings
-abrt_action_install_debuginfo_to_abrt_cache_LDADD = \
-     $(LIBREPORT_LIBS) \
-     ../lib/libabrt.la
-
 abrt_retrace_client_SOURCES = \
     abrt-retrace-client.c
  abrt_retrace_client_CFLAGS = \
diff --git a/src/plugins/abrt-action-list-dsos b/src/plugins/abrt-action-list-dsos
index 81a9927..bf1491c 100644
--- a/src/plugins/abrt-action-list-dsos
+++ b/src/plugins/abrt-action-list-dsos
@@ -5,7 +5,6 @@
 import sys
 import os
 import getopt
-import rpm
 
 def log(s):
     sys.stderr.write("%s\n" % s)
@@ -68,19 +67,10 @@ if __name__ == "__main__":
         try:
             dso_paths = parse_maps(memfile)
             for path in dso_paths:
-                ts = rpm.TransactionSet()
-                mi = ts.dbMatch('basenames', path)
-                if len(mi):
-                    for h in mi:
-                        if outname:
-                            outfile = xopen(outname, "w")
-                            outname = None
-                        outfile.write("%s %s (%s) %s\n" %
-                                    (path,
-                                     h[rpm.RPMTAG_NEVRA],
-                                     h[rpm.RPMTAG_VENDOR],
-                                     h[rpm.RPMTAG_INSTALLTIME])
-                                    )
+                if outname:
+                    outfile = xopen(outname, "w")
+                    outname = None
+                outfile.write(path)
 
         except Exception, ex:
             error_msg_and_die("Can't get the DSO list: %s" % ex)
diff --git a/src/plugins/ccpp_event.conf b/src/plugins/ccpp_event.conf
index 7d0d78c..5d4fa74 100644
--- a/src/plugins/ccpp_event.conf
+++ b/src/plugins/ccpp_event.conf
@@ -28,11 +28,10 @@ EVENT=collect_xsession_errors analyzer=CCpp dso_list~=.*/libX11.*
 # or was this ability lost with move to python installer?
 EVENT=analyze_LocalGDB analyzer=CCpp
         abrt-action-analyze-core --core=coredump -o build_ids &&
-        /usr/libexec/abrt-action-install-debuginfo-to-abrt-cache --size_mb=4096 &&
         abrt-action-generate-backtrace &&
         abrt-action-analyze-backtrace
 
 # Bugzilla requires nonempty duphash
-EVENT=report_Bugzilla analyzer=CCpp duphash!=
-        test -f component || abrt-action-save-package-data
-        reporter-bugzilla -f -b -c /etc/libreport/plugins/Bugzilla.conf
+# EVENT=report_Bugzilla analyzer=CCpp duphash!=
+#         test -f component || abrt-action-save-package-data
+#         reporter-bugzilla -f -b -c /etc/libreport/plugins/Bugzilla.conf
diff --git a/src/plugins/koops_event.conf b/src/plugins/koops_event.conf
index d203e1a..83842f9 100644
--- a/src/plugins/koops_event.conf
+++ b/src/plugins/koops_event.conf
@@ -13,5 +13,5 @@ EVENT=post-create analyzer=Kerneloops
 EVENT=report_Kerneloops analyzer=Kerneloops
         reporter-kerneloops
 
-EVENT=report_Bugzilla analyzer=Kerneloops
-        reporter-bugzilla -f -b
+# EVENT=report_Bugzilla analyzer=Kerneloops
+#         reporter-bugzilla -f -b
diff --git a/src/plugins/python_event.conf b/src/plugins/python_event.conf
index 16508c1..0cd10c6 100644
--- a/src/plugins/python_event.conf
+++ b/src/plugins/python_event.conf
@@ -1,6 +1,6 @@
 EVENT=post-create analyzer=Python
         abrt-action-analyze-python
 
-EVENT=report_Bugzilla analyzer=Python
-        test -f component || abrt-action-save-package-data
-        reporter-bugzilla -f -b -c /etc/libreport/plugins/Bugzilla.conf
+# EVENT=report_Bugzilla analyzer=Python
+#         test -f component || abrt-action-save-package-data
+#         reporter-bugzilla -f -b -c /etc/libreport/plugins/Bugzilla.conf
-- 
1.7.8.rc1

